src/invertedindex.js

```diff
@@ -0,0 +1,222 @@
+
+const hasProperty = Object.prototype.hasOwnProperty;
+let instance = null;
+
+class InvertedIndex {
+  /**
+    * constructor method
+  */
+  constructor() {
+    if (!instance) {
```

Comment:
The idea is to create a singleton class where object are created just once. I tested it and it worked perfectly well. I hope that worked well.